pkgs/collection/lib/src/iterable_extensions.dart

```diff
@@ -56,6 +57,99 @@ extension IterableExtension<T> on Iterable<T> {
     return chosen;
   }
 
+  /// The elements of this iterable separated by [separator].
+  ///
+  /// Emits the same elements as this iterable, and also emits
+  /// a [separator] between any two of those elements.
+  ///
+  /// If [before] is set to `true`, a [separator] is also
+  /// emitted before the first element.
+  /// If [after] is set to `true`, a [separator] is also
+  /// emitted after the last element.
+  ///
+  /// If this iterable is empty, [before] and [after] have no effect.
```

Comment:
Do you have any particular use cases in mind for the `before` and `after` arguments where not marking the boundaries of an empty iterable are the expected choice?

I don't necessarily doubt that this set of arguments/behaviors is the most useful one to provide, but I can't convince myself that it's the most useful behavior and it looks like it makes the Iterable implementation a fair bit more complex.